library/func_misc.py

```diff
@@ -153,3 +153,21 @@ def Call(self, pos_args, named_args):
 
         raise error.TypeErr(x, 'len() expected Str, List, or Dict',
                             loc.Missing)
+
+
+class Reverse(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+
+        r = typed_args.Reader(pos_args, named_args)
+        list_ = r.PosList()
+        r.Done()
+
+        list_.reverse()
+
+        return value.List(list_)
```

Comment:
Hm I actually think we should change this to value.Null to follow Python

```
>>> a=[1,2,3]
>>> print(a.reverse())
None
```

The reason is if people start using `a->reverse()` as a synonym for `reversed(a)`, they will get very confused by the resulting aliasing bugs (there's only one list, with two names, not two lists)


There is a spec test using that idiom

If we want that, we could add a reversed() wrapper that makes a copy, though I don't think it's 100% necessary now

i.e. I think Python's design of keeping mutation-in-place and transforming as separate APIs is a good one

Reply:
Gah, you're right. I see that I even made this error while updating a test. Returning `value.Null` is the way to go.